scripts/upload_artifacts.py

```diff
@@ -0,0 +1,127 @@
+import docker
+import sys
+import os
+import subprocess
+import yaml
+import boto3
+from botocore.exceptions import ClientError
+import logging
+
+# Configure logging
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+
+def run_command(command):
+    result = subprocess.run(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
+    if result.returncode != 0:
+        logging.error(f"Command failed: {command}\n{result.stderr}")
+        sys.exit(1)
+    return result.stdout
+
+def create_ecr_repository(repository_name, region):
+    session = boto3.Session(profile_name="devtest-admin")
+    client = session.client('ecr', region_name=region)
+    try:
+        response = client.create_repository(repositoryName=repository_name)
+        logging.info(f"Successfully created ECR repository: {repository_name}")
+        return response
+    except ClientError as e:
+        if e.response['Error']['Code'] == 'RepositoryAlreadyExistsException':
+            logging.info(f"{repository_name} repository already exists. Skipping.....")
+        else:
+            logging.error(f"Failed to create ECR repository: {repository_name}. Error: {e}")
+        return None
+
+def pull_and_push_images(image_list, destination_registry, region):
+    client = docker.from_env()
+
+    for image in image_list:
+        image_url = image["details"]["registryURL"].strip()
+        if not image_url:
+            continue
+
+        logging.info(f"Pulling image: {image_url}")
+        try:
+            pulled_image = client.images.pull(image_url)
+            logging.info(f"Successfully pulled image: {image_url}")
+        except Exception as e:
+            logging.error(f"Failed to pull image: {image_url}. Error: {e}")
+            continue
+
+        image_name = image_url.split('/')[-1].split(':')[0]
```

Comment:
That is actually most of the use-cases, it'll still work. We are only interested in the image name and tag. This logic only takes the last strings after the split which is the image name and tag regardless of the number of `/`

Reply:
I'm thinking of a situation where the image name is not unique even though the namespace is. Eg - There is a truefoundry image in dockerhub under `truefoundrycloud` and another within ecr

Reply:
What issue do you think this will cause?

Reply:
So two images - 
- `docker.io/truefoundrycloud/tfy-agent:latest`
- `public.ecr.aws.com/truefoundrycloud/tfy-agent:latest`
These could be completely different images since latest can be different.

When we push to our own registry, both will end up having url - 
- `<custom_registry>/tfy-agent:latest` which can be a problem

Also the kyverno patch only replaces the first part with the registry url. So `docker.io/truefoundrycloud/tfy-agent:latest` becomes `<custom_registry>/truefoundrycloud/tfy-agent:latest`